### PR TITLE
Add Theme3 addon

### PR DESCRIPTION
**Pull request by ZenithRogue**
### **Changes**
Added block color theming
image

Color inputs have also been expanded, with a clear button added as well

### **Tests**
This didn't originally work with colored context menus, but with a small change to that addon, I was able to fix that as well.
image

### CODE REVIEW

**Comment by GrahamSH-LLK** [comment]
Huh. Annoying that it's not automatically linting. Anyway, once it's merged it will.

File: addons/editor-theme3/theme3.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        	}`;
+    }
+    if (prop == "events") {
+      stylesheet += `path.blocklyBlockBackground[fill="#FFBF00"] {
+				fill: var(--editorTheme3-${prop}Color);
+        	}`;
+    }
+  }
+
+  style.innerHTML = stylesheet;
```
Comment (GrahamSH-LLK) [inline addons/editor-theme3/theme3.js:102]: Can you use something other than innerHTML?

**Review by GrahamSH-LLK — CHANGES_REQUESTED** [review]

File: addons/editor-theme3/theme3.js (hunk 1 above)
Comment (WorldLanguages) [reply addons/editor-theme3/theme3.js:102]: textContent / innerText would work just fine here

**Comment by WorldLanguages** [comment]
@ZenithRogue, on commit e0e62a3 I moved non-dynamic CSS to an userstyle. I also made it so the dynamic CSS gets added inside the same \<style> element as the non-dynamic CSS, to avoid issues with style priority - it's not a very good idea to make style priority depend on the order in which JavaScript runs.

**Comment by WorldLanguages** [comment]
@ZenithRogue Hmm do you think this addon deserves a more descriptive name and description? Also, don't forget descriptions need a dot at the end (lol)

**Comment by JeffaloBob** [comment]
that addon looks absolutely amazing woah!!!!! 

perhaps it should be called something more generic like w_l said. what does the "randomize" checkmark in your screenshot do?

**Comment by ZenithRogue** [comment]
@JeffaloBob the randomize check box will randomly choose colors every time you open the editor, without changing your selected colors

**Comment by WorldLanguages** [comment]
@ZenithRogue I'd prefer not to use already existing names unless the previous extension/userscript/etc was relatively popular under the same name. If you really want "Theme3" to be within the addon name we can discuss it after merging.

**Comment by ZenithRogue** [comment]
> @ZenithRogue, on commit [e0e62a3](https://github.com/ScratchAddons/ScratchAddons/commit/e0e62a3dcd85d92ce0cbd912ddeda51cabd411ed) I moved non-dynamic CSS to an userstyle. I also made it so the dynamic CSS gets added inside the same <style> element as the non-dynamic CSS, to avoid issues with style priority - it's not a very good idea to make style priority depend on the order in which JavaScript runs.

@WorldLanguages I just realized that this change broke the dynamic colors, preventing you from live-editing your block colors

**Comment by WorldLanguages** [comment]
@ZenithRogue Weird, I thought I made CSS variables update automatically? I might have forgotten

**Comment by ZenithRogue** [comment]
I'm not sure exactly what causes the break with your change. The style element seems to vanish entirely if you update any of the settings.

**Comment by WorldLanguages** [comment]
I can't reproduce, @ZenithRogue. I can edit the block colors normally, and it updates real time.

**Comment by Explosion-Scratch** [comment]
Works for me too. What browser/os are you using?

**Comment by Explosion-Scratch** [comment]
I'm on chrome (latest) on chromebook.

**Pull request merged**